yogstation/code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -0,0 +1,272 @@
+	reports of odd or suspicious sightings in maintenance."
+
+/datum/game_mode/shadowling/post_setup()
+	for(var/datum/mind/shadow in shadows)
+		log_game("[shadow.key] (ckey) has been selected as a Shadowling.")
+		shadow.current.add_sling()
+	. = ..()
+	return
+
+/datum/game_mode/shadowling/proc/check_shadow_victory()
+	return shadowling_ascended
+
+/datum/game_mode/shadowling/proc/check_shadow_death()
+	for(var/SM in get_antag_minds(/datum/antagonist/shadowling))
+		var/datum/mind/shadow_mind = SM
+		if(istype(shadow_mind))
+			var/turf/T = get_turf(shadow_mind.current)
+			if((shadow_mind) && (shadow_mind.current) && (shadow_mind.current.stat != DEAD) && T && is_station_level(T.z) && ishuman(shadow_mind.current))
+				return FALSE
+	return TRUE
+
+/datum/game_mode/shadowling/check_finished()
+	. = ..()
+	if(check_shadow_death())
+		return TRUE
+
+
+/datum/game_mode/proc/auto_declare_completion_shadowling()
+	var/text = ""
+	if(shadows.len)
+		text += "<br><span class='big'><b>The shadowlings were:</b></span>"
+		for(var/datum/mind/shadow in shadows)
+			text += printplayer(shadow)
+		text += "<br>"
+		if(thralls.len)
+			text += "<br><span class='big'><b>The thralls were:</b></span>"
+			for(var/datum/mind/thrall in thralls)
+				text += printplayer(thrall)
+	text += "<br>"
+	to_chat(world, text)
+
+
+/*
+	MISCELLANEOUS
+*/
+
+/datum/species/shadow/ling
+	//Normal shadowpeople but with enhanced effects
+	name = "Shadowling"
+	id = "shadowling"
+	say_mod = "chitters"
+	species_traits = list(NOBLOOD,NO_UNDERWEAR,NO_DNA_COPY,NOTRANSSTING,NOEYES)
+	inherent_traits = list(TRAIT_NOGUNS, TRAIT_RESISTCOLD, TRAIT_RESISTHIGHPRESSURE,TRAIT_RESISTLOWPRESSURE, TRAIT_NOBREATH, TRAIT_RADIMMUNE, TRAIT_VIRUSIMMUNE, TRAIT_PIERCEIMMUNE)
+	no_equip = list(SLOT_WEAR_MASK, SLOT_GLASSES, SLOT_GLOVES, SLOT_SHOES, SLOT_W_UNIFORM, SLOT_S_STORE)
+	nojumpsuit = 1
+	mutanteyes = /obj/item/organ/eyes/night_vision/alien/sling
+	burnmod = 1.5 //1.5x burn damage, 2x is excessive
+	heatmod = 1.5
+	var/mutable_appearance/eyes_overlay
+	var/shadow_charges = 3
+	var/last_charge = 0
+
+/datum/species/shadow/ling/on_species_gain(mob/living/carbon/human/C)
+	C.draw_yogs_parts()
+	eyes_overlay = mutable_appearance('yogstation/icons/mob/sling.dmi', "eyes", 25)
+	C.add_overlay(eyes_overlay)
+	. = ..()
+
+/datum/species/shadow/ling/on_species_loss(mob/living/carbon/human/C)
+	C.draw_yogs_parts(TRUE)
+	if(eyes_overlay)
+		C.cut_overlay(eyes_overlay)
+		QDEL_NULL(eyes_overlay)
+	. = ..()
+
+/datum/species/shadow/ling/spec_life(mob/living/carbon/human/H)
+	var/light_amount = 0
+	H.nutrition = NUTRITION_LEVEL_WELL_FED //i aint never get hongry
+	if(isturf(H.loc))
+		var/turf/T = H.loc
+		light_amount = T.get_rgb_lumcount(r_mul = LIGHT_RED_MULTIPLIER, g_mul = LIGHT_GREEN_MULTIPLIER, b_mul = LIGHT_BLUE_MULTIPLIER)
```

Comment:
Delete the ``var/light_amount = 0`` at the start of the proc and make this ``var/light_amount = ...``.